File: solutions/java/0069_x的平方根.java
package com.leetcode;

public class Solution {
    public int mySqrt(int x) {
        if (x < 2) return x;
        long left = 1, right = x / 2;
        while (left <= right) {
            long mid = (left + right) / 2;
            long sq = mid * mid;
            if (sq == x) return (int) mid;
            else if (sq < x) left = mid + 1;
            else right = mid - 1;
        }
        return (int) right;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[][] cases = {{4, 2}, {8, 2}, {0, 0}, {1, 1}, {9, 3}};
        for (int[] tc : cases) {
            int result = sol.mySqrt(tc[0]);
            String ok = result == tc[1] ? "PASS" : "FAIL";
            System.out.printf("mySqrt(%d) = %d, expected %d: %s%n", tc[0], result, tc[1], ok);
        }
        // 大数测试
        int big = sol.mySqrt(2147395600);
        System.out.printf("mySqrt(2147395600) = %d, expected 46340: %s%n", big, big == 46340 ? "PASS" : "FAIL");
    }
}

File: solutions/python/0069_x的平方根.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x < 2:
            return x
        left, right = 1, x // 2
        while left <= right:
            mid = (left + right) // 2
            sq = mid * mid
            if sq == x:
                return mid
            elif sq < x:
                left = mid + 1
            else:
                right = mid - 1
        return right

if __name__ == "__main__":
    test_cases = [
        (4, 2),
        (8, 2),
        (0, 0),
        (1, 1),
        (9, 3),
        (2147395600, 46340),
    ]
    sol = Solution()
    for x, expected in test_cases:
        result = sol.mySqrt(x)
        ok = result == expected
        print(f"mySqrt({x}) = {result}, expected {expected}: {'PASS' if ok else 'FAIL'}")
